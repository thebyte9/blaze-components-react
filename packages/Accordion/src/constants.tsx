const FLEX = "flex";
const NONE = "none";
const DOWN = "down";
const UP = "up";

export { FLEX, NONE, DOWN, UP };
