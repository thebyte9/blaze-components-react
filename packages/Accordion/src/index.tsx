import React, { useEffect, useState } from "react";
import AccordionContent from "./AccordionContent";
import AccordionContentDetails from "./AccordionContent/AccordionContentDetails";
import AccordionContentFooter from "./AccordionContent/AccordionContentFooter";
import AccordionHeader from "./AccordionHeader";

interface IAccordionProps {
  children: [JSX.Element, JSX.Element];
  open?: boolean;
}

const Accordion = ({ children, open }: IAccordionProps): JSX.Element => {
  const flex = "flex";
  const none = "none";

  const [accordionStatus, setAccordionStatus] = useState<string>(
    open ? flex : none
  );

  const [header, content]: [JSX.Element, JSX.Element] = children;

  useEffect(() => {
    setAccordionStatus(open ? flex : none);
  }, [open]);

  const isActive: boolean = accordionStatus === flex;
  const toggleAccordion = (): void =>
    setAccordionStatus(isActive ? none : flex);

  return (
    <div className="accordion">
      <div className="accordion__header">
        {header}
        <div
          className="icon-button icon-button--accordion"
          onClick={toggleAccordion}
        >
          <i className="material-icons">keyboard_arrow_down</i>
        </div>
      </div>
      <div
        className="accordion__content-wrapper"
        style={{ display: accordionStatus }}
      >
        {content}
      </div>
    </div>
  );
};

Accordion.Content = AccordionContent;
Accordion.ContentDetails = AccordionContentDetails;
Accordion.ContentFooter = AccordionContentFooter;
Accordion.Header = AccordionHeader;
export default Accordion;
