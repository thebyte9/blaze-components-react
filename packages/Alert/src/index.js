import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
import Button from '@blaze-react/button';

const Alert = ({ children, close, icon, type, ...attrs }) => {
  const [offModal, setModalOff] = useState(false);
  const assignType = type && `alert--${type}`;
  const isDismissable = close && 'alert--dismissable';
  const withIcon = icon && 'alert--icon';

  const renderAlert = (
    <div className={`alert ${assignType} ${isDismissable} ${withIcon}`} {...attrs}>
      {icon && <i className="material-icons">{icon}</i>}
      {children}
      {close && (
        <Button onClick={() => setModalOff(true)} className="icon-button icon-button--close">
          <i className="material-icons">close</i>
        </Button>
      )}
    </div>
  );

  return <Fragment>{!offModal && renderAlert}</Fragment>;
};

Alert.propTypes = {
  icon: PropTypes.string,
  type: PropTypes.string,
  close: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

Alert.defaultProps = {
  icon: '',
  type: '',
  close: false,
  children: 'No content'
};

export default Alert;
