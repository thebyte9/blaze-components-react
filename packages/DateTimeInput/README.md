## Description

Datepicker tools that wraps [react-datepicker](https://www.npmjs.com/package/react-datepicker).


## Note about importing

This component requires its consumers to manually import the third-party library's CSS

```typescript
import "react-datepicker/dist/react-datepicker.css";
```


## Usage

- Required

```js
<DateTimeInput
  label="Date Time input required"
  onChange={({ event, value }) => {}}
  type="dateTime"
  modifier="full-width"
  validationMessage="Date Time is required"
  error
  required
/>
```

- Disabled

```js
<DateTimeInput
  label="Date Time input disabled"
  onChange={({ event, value }) => {}}
  type="dateTime"
  modifier="full-width"
  validationMessage="Date Time is disabled"
  error
  disabled
/>
```

- Pick only date

```js
<DateTimeInput
  label="Date input"
  onChange={({ event, value }) => {}}
  type="date"
  modifier="full-width"
/>
```

- Pick only time

```js
<DateTimeInput
  label="Date input"
  onChange={({ event, value }) => {}}
  type="time"
  modifier="full-width"
/>
```

## API

##### DateTimeInput can receive a number of `props` as follow:

| NAME              |   TYPE                        |   DEFAULT              |
| :---------------- | :---------------------------: | :--------------------: |
| label             |            String             |       empty            |
| type              | ['dateTime', 'date', 'time']  |      dateTime          |
| value             |            String             |         empty          |
| validationMessage |            String             | This field is required |
| disabled          |           Boolean             |         false          |
| required          |           Boolean             |         false          |
| error             |           Boolean             |         false          |
| onChange          |           Function            |        () => {}        |
