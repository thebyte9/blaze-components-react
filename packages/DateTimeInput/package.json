{
  "name": "@blaze-react/date-time-input",
  "version": "0.8.0-alpha.36",
  "description": "",
  "main": "./dist/main.js",
  "scripts": {
    "build": "NODE_ENV=production webpack --config webpack.config.js",
    "build-storybook": "build-storybook -c .storybook -o docs-build",
    "lint": "eslint '*/**/*.{ts,tsx}' --ignore-path ../../.eslintignore --fix",
    "test": "jest --passWithNoTests",
    "test:coverage": "jest --coverage --passWithNoTests",
    "pre-publish": "yarn build && yarn lint && yarn test",
    "bundle-analyser": "node ../../node_modules/webpack-bundle-analyzer/lib/bin/analyzer.js ./stats/stats.json"
  },
  "keywords": [],
  "author": "byte9",
  "license": "GPL-3.0",
  "devDependencies": {
    "@babel/cli": "^7.13.14",
    "@babel/core": "^7.13.14",
    "@types/react-datepicker": "^3.1.8"
  },
  "dependencies": {
    "react-datepicker": "^3.7.0",
    "webpack": "^5.28.0",
    "webpack-cli": "^4.6.0"
  },
  "gitHead": "4bfe5dcb677b6f46229be4039ba4ca7db86e7cca"
}
