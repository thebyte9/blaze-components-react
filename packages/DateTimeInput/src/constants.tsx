const TYPE_DATE = "date";
const TYPE_DATE_TIME = "dateTime";
const TYPE_TIME = "time";

export { TYPE_DATE, TYPE_DATE_TIME, TYPE_TIME };
