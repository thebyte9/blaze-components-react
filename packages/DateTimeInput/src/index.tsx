import withUtils from "@blaze-react/utils";
import React, { FunctionComponent, useEffect, useState, useRef } from "react";
import DatePicker from "react-datepicker";
import {
  DATE_FORMAT_MAP,
  TYPE_DATE,
  TYPE_DATE_TIME,
  TYPE_TIME,
} from "./constants";

// TODO We are not sure about whether this import will work in any context
// import "react-datepicker/dist/react-datepicker.css";

interface IErrorMessage {
  message: string | JSX.Element;
  icon?: string;
}

interface IDateTimeInputProps {
  disabled?: boolean;
  id?: string;
  label?: string;
  modifier?: string;
  onChange: ({
    event,
    value,
  }: {
    event:
      | React.ChangeEvent<HTMLInputElement>
      | React.MouseEvent<HTMLDivElement, MouseEvent>;
    value: Date;
  }) => void;
  required?: boolean;
  error?: boolean;
  type?: string;
  validationMessage: string | JSX.Element;
  value?: Date;
  utils: {
    buildClassNames: (
      className: string | object,
      optionalClassNames?: object
    ) => string;
    ErrorMessage: FunctionComponent<IErrorMessage>;
  };
}

const DateTimeInput: FunctionComponent<IDateTimeInputProps> = ({
  disabled,
  id,
  label,
  modifier,
  onChange,
  required,
  error,
  type,
  validationMessage,
  value,
  utils: { buildClassNames, ErrorMessage },
}): JSX.Element => {
  const [newValue, setNewValue] = useState<Date | undefined>(value);
  const [newError, setError] = useState<boolean | undefined>(error);
  const [newOpen, setOpen] = useState<boolean>(false);
  const containerRef = useRef<any>(null);

  let whitelistedType: string = type as string;
  if (type !== TYPE_TIME && type !== TYPE_DATE) {
    whitelistedType = TYPE_DATE_TIME;
  }

  useEffect(() => {
    if (newOpen && containerRef && containerRef.current) {
      const selectedDay = containerRef.current.querySelector(
        ".react-datepicker__day--selected"
      );
      if (selectedDay) {
        selectedDay.focus();
      }
    }
  }, [newOpen, containerRef]);

  useEffect(() => {
    setError(error);
  }, [error]);

  useEffect(() => {
    setNewValue(value);
  }, [value]);

  const handleChange = (
    date: Date,
    event: any,
    forceClose: boolean = false
  ): void => {
    setNewValue(date);
    setOpen(!forceClose);

    // make event consistent
    const { target = {} } = event || {};
    target.name = id;
    target.value = date;

    // FIXME the native event is not always returned by react-datepicker
    // See https://byte-9.atlassian.net/browse/BZ2-2130
    onChange({ event: { ...event, target }, value: date });
  };

  const requiredClassName: string = buildClassNames({ required });

  const rootClasses: string = buildClassNames(
    "form-field form-field--date-time-input",
    {
      [`form-field--${modifier}`]: !!modifier,
    }
  );

  return (
    <div className={rootClasses} ref={containerRef}>
      <label htmlFor={id} className={requiredClassName}>
        {label}
      </label>

      <DatePicker
        data-testid="date-time-input"
        id={id}
        name={id}
        onChange={handleChange}
        onFocus={() => setOpen(true)}
        showTimeInput={[TYPE_TIME, TYPE_DATE_TIME].includes(whitelistedType)}
        showTimeSelectOnly={whitelistedType === TYPE_TIME}
        onClickOutside={() => setOpen(false)}
        popperModifiers={{
          flip: {
            enabled: false,
          },
        }}
        dateFormat={DATE_FORMAT_MAP[whitelistedType]}
        open={newOpen}
        isClearable={true}
        selected={newValue}
        disabled={disabled}
        required={required}
      >
        {(whitelistedType === TYPE_DATE_TIME ||
          whitelistedType === TYPE_DATE) && (
          <div
            className="react-datepicker__today-button"
            onClick={(e) => handleChange(new Date(), e, true)}
          >
            Today
          </div>
        )}
      </DatePicker>

      {newError && <ErrorMessage message={validationMessage} />}
    </div>
  );
};

DateTimeInput.defaultProps = {
  disabled: false,
  error: false,
  label: "",
  modifier: "",
  required: false,
  type: TYPE_DATE_TIME,
  validationMessage: "This field is required",
};

export default withUtils(DateTimeInput);
