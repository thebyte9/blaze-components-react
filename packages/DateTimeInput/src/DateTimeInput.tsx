import { buildClassNames } from '@blaze-react/utils';
import { ErrorMessage } from '@blaze-react/utils';
import React, { FunctionComponent, useEffect, useState } from 'react';
import DatePicker from 'react-datepicker';
import { DATE_FORMAT_MAP, TYPE_DATE, TYPE_DATE_TIME, TYPE_TIME } from './constants';

// TODO We are not sure about whether this import will work in any context
// import "react-datepicker/dist/react-datepicker.css";

interface IDateTimeInputProps {
  disabled?: boolean;
  id?: string;
  label?: string;
  modifier?: string;
  onChange: ({
    event,
    value,
  }: {
    event: React.ChangeEvent<HTMLInputElement> | React.MouseEvent<HTMLDivElement, MouseEvent>;
    value: Date;
  }) => void;
  required?: boolean;
  error?: boolean;
  type?: string;
  validationMessage: string | JSX.Element;
  value?: Date;
}

const DateTimeInput: FunctionComponent<IDateTimeInputProps> = ({
  disabled,
  id,
  label,
  modifier,
  onChange,
  required,
  error,
  type,
  validationMessage,
  value,
}): JSX.Element => {
  const [newValue, setNewValue] = useState<Date | undefined>(value);
  const [newError, setError] = useState<boolean | undefined>(error);
  const [newOpen, setOpen] = useState<boolean>(false);

  let whitelistedType = '';

  if (type !== TYPE_TIME && type !== TYPE_DATE) {
    whitelistedType = TYPE_DATE_TIME;
  }

  useEffect(() => {
    setError(error);
  }, [error]);

  useEffect(() => {
    setNewValue(value);
  }, [value]);

  const handleChange = (date: Date, event: any, forceClose = false): void => {
    setNewValue(date);
    setOpen(!forceClose);

    // make event consistent
    const { target = {} } = event || {};
    target.name = id;
    target.value = date;

    // FIXME the native event is not always returned by react-datepicker
    // See https://byte-9.atlassian.net/browse/BZ2-2130
    onChange({ event: { ...event, target }, value: date });
  };

  const requiredClassName: string = buildClassNames({ required });

  const rootClasses: string = buildClassNames('form-field form-field--date-time-input', {
    [`form-field--${modifier}`]: !!modifier,
  });

  return (
    <div className={rootClasses}>
      <label htmlFor={id} className={requiredClassName}>
        {label}
      </label>

      <DatePicker
        data-testid={id}
        id={id}
        name={id}
        onChange={handleChange}
        onFocus={() => setOpen(true)}
        showTimeInput={[TYPE_TIME, TYPE_DATE_TIME].includes(whitelistedType)}
        showTimeSelectOnly={whitelistedType === TYPE_TIME}
        onClickOutside={() => setOpen(false)}
        popperModifiers={{
          flip: {
            enabled: false,
          },
        }}
        dateFormat={DATE_FORMAT_MAP[whitelistedType]}
        open={newOpen}
        isClearable={true}
        selected={newValue}
        disabled={disabled}
        required={required}
      >
        {(whitelistedType === TYPE_DATE_TIME || whitelistedType === TYPE_DATE) && (
          <div className="react-datepicker__today-button" onClick={(e) => handleChange(new Date(), e, true)}>
            Today
          </div>
        )}
      </DatePicker>

      {newError && <ErrorMessage message={validationMessage} />}
    </div>
  );
};

DateTimeInput.defaultProps = {
  disabled: false,
  error: false,
  label: '',
  modifier: '',
  required: false,
  type: TYPE_DATE_TIME,
  validationMessage: 'This field is required',
};

export default DateTimeInput;
