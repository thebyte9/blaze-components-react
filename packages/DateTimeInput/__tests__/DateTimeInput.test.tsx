import { mount } from "enzyme";
import "jest-dom/extend-expect";
import React from "react";
import DateTimeInput from "../src/DateTimeInput";
import { TYPE_DATE, TYPE_DATE_TIME, TYPE_TIME } from '../src/constants';

const defaultProps = (override: object = {}) => ({
  error: true,
  onChange: () => void 0,
  type: TYPE_DATE_TIME,
  ...override,
});

describe("DateTimeInput component", () => {
  test("should be defined and renders correctly type=dateTime (snapshot)", () => {
    const wrapper = mount(<DateTimeInput {...defaultProps()} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test("should be defined and renders correctly type=date (snapshot)", () => {
    const wrapper = mount(<DateTimeInput {...defaultProps({ type: TYPE_DATE })} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test("should be defined and renders correctly type=time (snapshot)", () => {
    const wrapper = mount(<DateTimeInput {...defaultProps({ type: TYPE_TIME })} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test("should be defined and renders correctly type=undefined (snapshot)", () => {
    const wrapper = mount(<DateTimeInput {...defaultProps({ type: undefined })}/>);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
