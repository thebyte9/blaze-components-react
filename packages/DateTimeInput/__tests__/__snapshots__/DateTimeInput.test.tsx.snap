// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimeInput component should be defined and renders correctly type=date (snapshot) 1`] = `
<withUtils(DateTimeInput)
  error={true}
  onChange={[Function]}
  type="date"
>
  <DateTimeInput
    disabled={false}
    error={true}
    label=""
    modifier=""
    onChange={[Function]}
    required={false}
    type="date"
    utils={
      Object {
        "ErrorMessage": [Function],
        "buildClassNames": [Function],
        "removeExtraSpaces": [Function],
        "uniqueId": [Function],
      }
    }
    validationMessage="This field is required"
  >
    <div
      className="form-field form-field--date-time-input"
    >
      <label
        className=""
      />
      <r
        allowSameDay={false}
        customTimeInput={null}
        data-testid="date-time-input"
        dateFormat="MMMM d, yyyy"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        enableTabLoop={true}
        excludeScrollbar={true}
        focusSelectedMonth={false}
        isClearable={true}
        monthsShown={1}
        nextMonthButtonLabel="Next Month"
        nextYearButtonLabel="Next Year"
        onBlur={[Function]}
        onCalendarClose={[Function]}
        onCalendarOpen={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        open={false}
        popperModifiers={
          Object {
            "flip": Object {
              "enabled": false,
            },
          }
        }
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        previousYearButtonLabel="Previous Year"
        readOnly={false}
        renderDayContents={[Function]}
        required={false}
        shouldCloseOnSelect={true}
        showFourColumnMonthYearPicker={false}
        showFullMonthYearPicker={false}
        showMonthYearPicker={false}
        showPopperArrow={true}
        showPreviousMonths={false}
        showQuarterYearPicker={false}
        showTimeInput={false}
        showTimeSelect={false}
        showTimeSelectOnly={false}
        showTwoColumnMonthYearPicker={false}
        showYearPicker={false}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
        yearItemNumber={12}
      >
        <r
          enableTabLoop={true}
          hidePopper={true}
          popperComponent={null}
          popperModifiers={
            Object {
              "flip": Object {
                "enabled": false,
              },
            }
          }
          popperOnKeyDown={[Function]}
          popperPlacement="bottom-start"
          popperProps={Object {}}
          targetComponent={
            <div
              className="react-datepicker__input-container"
            >
              <input
                className=""
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                readOnly={false}
                required={false}
                type="text"
                value=""
              />
            </div>
          }
        >
          <Manager
            className="react-datepicker-manager"
          >
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <div
                  className="react-datepicker-wrapper"
                >
                  <div
                    className="react-datepicker__input-container"
                  >
                    <input
                      className=""
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      readOnly={false}
                      required={false}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </InnerReference>
            </Reference>
          </Manager>
        </r>
      </r>
      <ErrorMessage
        icon="warning"
        message="This field is required"
      >
        <div
          className="validation"
          data-testid="validation-message"
        >
          <i
            className="material-icons"
          >
            warning
          </i>
          This field is required
        </div>
      </ErrorMessage>
    </div>
  </DateTimeInput>
</withUtils(DateTimeInput)>
`;

exports[`DateTimeInput component should be defined and renders correctly type=dateTime (snapshot) 1`] = `
<withUtils(DateTimeInput)
  error={true}
  onChange={[Function]}
  type="dateTime"
>
  <DateTimeInput
    disabled={false}
    error={true}
    label=""
    modifier=""
    onChange={[Function]}
    required={false}
    type="dateTime"
    utils={
      Object {
        "ErrorMessage": [Function],
        "buildClassNames": [Function],
        "removeExtraSpaces": [Function],
        "uniqueId": [Function],
      }
    }
    validationMessage="This field is required"
  >
    <div
      className="form-field form-field--date-time-input"
    >
      <label
        className=""
      />
      <r
        allowSameDay={false}
        customTimeInput={null}
        data-testid="date-time-input"
        dateFormat="MMMM d, yyyy HH:mm"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        enableTabLoop={true}
        excludeScrollbar={true}
        focusSelectedMonth={false}
        isClearable={true}
        monthsShown={1}
        nextMonthButtonLabel="Next Month"
        nextYearButtonLabel="Next Year"
        onBlur={[Function]}
        onCalendarClose={[Function]}
        onCalendarOpen={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        open={false}
        popperModifiers={
          Object {
            "flip": Object {
              "enabled": false,
            },
          }
        }
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        previousYearButtonLabel="Previous Year"
        readOnly={false}
        renderDayContents={[Function]}
        required={false}
        shouldCloseOnSelect={true}
        showFourColumnMonthYearPicker={false}
        showFullMonthYearPicker={false}
        showMonthYearPicker={false}
        showPopperArrow={true}
        showPreviousMonths={false}
        showQuarterYearPicker={false}
        showTimeInput={true}
        showTimeSelect={false}
        showTimeSelectOnly={false}
        showTwoColumnMonthYearPicker={false}
        showYearPicker={false}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
        yearItemNumber={12}
      >
        <r
          enableTabLoop={true}
          hidePopper={true}
          popperComponent={null}
          popperModifiers={
            Object {
              "flip": Object {
                "enabled": false,
              },
            }
          }
          popperOnKeyDown={[Function]}
          popperPlacement="bottom-start"
          popperProps={Object {}}
          targetComponent={
            <div
              className="react-datepicker__input-container"
            >
              <input
                className=""
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                readOnly={false}
                required={false}
                type="text"
                value=""
              />
            </div>
          }
        >
          <Manager
            className="react-datepicker-manager"
          >
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <div
                  className="react-datepicker-wrapper"
                >
                  <div
                    className="react-datepicker__input-container"
                  >
                    <input
                      className=""
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      readOnly={false}
                      required={false}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </InnerReference>
            </Reference>
          </Manager>
        </r>
      </r>
      <ErrorMessage
        icon="warning"
        message="This field is required"
      >
        <div
          className="validation"
          data-testid="validation-message"
        >
          <i
            className="material-icons"
          >
            warning
          </i>
          This field is required
        </div>
      </ErrorMessage>
    </div>
  </DateTimeInput>
</withUtils(DateTimeInput)>
`;

exports[`DateTimeInput component should be defined and renders correctly type=time (snapshot) 1`] = `
<withUtils(DateTimeInput)
  error={true}
  onChange={[Function]}
  type="time"
>
  <DateTimeInput
    disabled={false}
    error={true}
    label=""
    modifier=""
    onChange={[Function]}
    required={false}
    type="time"
    utils={
      Object {
        "ErrorMessage": [Function],
        "buildClassNames": [Function],
        "removeExtraSpaces": [Function],
        "uniqueId": [Function],
      }
    }
    validationMessage="This field is required"
  >
    <div
      className="form-field form-field--date-time-input"
    >
      <label
        className=""
      />
      <r
        allowSameDay={false}
        customTimeInput={null}
        data-testid="date-time-input"
        dateFormat="HH:mm"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        enableTabLoop={true}
        excludeScrollbar={true}
        focusSelectedMonth={false}
        isClearable={true}
        monthsShown={1}
        nextMonthButtonLabel="Next Month"
        nextYearButtonLabel="Next Year"
        onBlur={[Function]}
        onCalendarClose={[Function]}
        onCalendarOpen={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        open={false}
        popperModifiers={
          Object {
            "flip": Object {
              "enabled": false,
            },
          }
        }
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        previousYearButtonLabel="Previous Year"
        readOnly={false}
        renderDayContents={[Function]}
        required={false}
        shouldCloseOnSelect={true}
        showFourColumnMonthYearPicker={false}
        showFullMonthYearPicker={false}
        showMonthYearPicker={false}
        showPopperArrow={true}
        showPreviousMonths={false}
        showQuarterYearPicker={false}
        showTimeInput={true}
        showTimeSelect={false}
        showTimeSelectOnly={true}
        showTwoColumnMonthYearPicker={false}
        showYearPicker={false}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
        yearItemNumber={12}
      >
        <r
          enableTabLoop={true}
          hidePopper={true}
          popperComponent={null}
          popperModifiers={
            Object {
              "flip": Object {
                "enabled": false,
              },
            }
          }
          popperOnKeyDown={[Function]}
          popperPlacement="bottom-start"
          popperProps={Object {}}
          targetComponent={
            <div
              className="react-datepicker__input-container"
            >
              <input
                className=""
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                readOnly={false}
                required={false}
                type="text"
                value=""
              />
            </div>
          }
        >
          <Manager
            className="react-datepicker-manager"
          >
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <div
                  className="react-datepicker-wrapper"
                >
                  <div
                    className="react-datepicker__input-container"
                  >
                    <input
                      className=""
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      readOnly={false}
                      required={false}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </InnerReference>
            </Reference>
          </Manager>
        </r>
      </r>
      <ErrorMessage
        icon="warning"
        message="This field is required"
      >
        <div
          className="validation"
          data-testid="validation-message"
        >
          <i
            className="material-icons"
          >
            warning
          </i>
          This field is required
        </div>
      </ErrorMessage>
    </div>
  </DateTimeInput>
</withUtils(DateTimeInput)>
`;

exports[`DateTimeInput component should be defined and renders correctly type=undefined (snapshot) 1`] = `
<withUtils(DateTimeInput)
  error={true}
  onChange={[Function]}
>
  <DateTimeInput
    disabled={false}
    error={true}
    label=""
    modifier=""
    onChange={[Function]}
    required={false}
    type="dateTime"
    utils={
      Object {
        "ErrorMessage": [Function],
        "buildClassNames": [Function],
        "removeExtraSpaces": [Function],
        "uniqueId": [Function],
      }
    }
    validationMessage="This field is required"
  >
    <div
      className="form-field form-field--date-time-input"
    >
      <label
        className=""
      />
      <r
        allowSameDay={false}
        customTimeInput={null}
        data-testid="date-time-input"
        dateFormat="MMMM d, yyyy HH:mm"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        enableTabLoop={true}
        excludeScrollbar={true}
        focusSelectedMonth={false}
        isClearable={true}
        monthsShown={1}
        nextMonthButtonLabel="Next Month"
        nextYearButtonLabel="Next Year"
        onBlur={[Function]}
        onCalendarClose={[Function]}
        onCalendarOpen={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        open={false}
        popperModifiers={
          Object {
            "flip": Object {
              "enabled": false,
            },
          }
        }
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        previousYearButtonLabel="Previous Year"
        readOnly={false}
        renderDayContents={[Function]}
        required={false}
        shouldCloseOnSelect={true}
        showFourColumnMonthYearPicker={false}
        showFullMonthYearPicker={false}
        showMonthYearPicker={false}
        showPopperArrow={true}
        showPreviousMonths={false}
        showQuarterYearPicker={false}
        showTimeInput={true}
        showTimeSelect={false}
        showTimeSelectOnly={false}
        showTwoColumnMonthYearPicker={false}
        showYearPicker={false}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
        yearItemNumber={12}
      >
        <r
          enableTabLoop={true}
          hidePopper={true}
          popperComponent={null}
          popperModifiers={
            Object {
              "flip": Object {
                "enabled": false,
              },
            }
          }
          popperOnKeyDown={[Function]}
          popperPlacement="bottom-start"
          popperProps={Object {}}
          targetComponent={
            <div
              className="react-datepicker__input-container"
            >
              <input
                className=""
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                readOnly={false}
                required={false}
                type="text"
                value=""
              />
            </div>
          }
        >
          <Manager
            className="react-datepicker-manager"
          >
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <div
                  className="react-datepicker-wrapper"
                >
                  <div
                    className="react-datepicker__input-container"
                  >
                    <input
                      className=""
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      readOnly={false}
                      required={false}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </InnerReference>
            </Reference>
          </Manager>
        </r>
      </r>
      <ErrorMessage
        icon="warning"
        message="This field is required"
      >
        <div
          className="validation"
          data-testid="validation-message"
        >
          <i
            className="material-icons"
          >
            warning
          </i>
          This field is required
        </div>
      </ErrorMessage>
    </div>
  </DateTimeInput>
</withUtils(DateTimeInput)>
`;
